src/dbcp/transform/gridstatus.py

```diff
@@ -0,0 +1,632 @@
+    )
+    iso_df = _create_project_status_classification_from_single_column(
+        iso_df,
+        "Status (Original)",
+        nearly_certain_cols,
+        actionable_cols,
+    )
+    return iso_df
+
+
+def _transform_nyiso(iso_df: pd.DataFrame) -> pd.DataFrame:
+    """Make nyiso specific transformations.
+
+    NYISO Status Key:
+        * 0=Withdrawn
+        * 1=Scoping Meeting Pending
+        * 2=FES Pending
+        * 3=FES in Progress
+        * 4=SRIS/SIS Pending
+        * 5=SRIS/SIS in Progress
+        * 6=SRIS/SIS Approved
+        * 7=FS Pending
+        * 8=Rejected Cost Allocation/Next FS Pending
+        * 9=FS in Progress
+        * 10=Accepted Cost Allocation/IA in Progress
+        * 11=IA Completed
+        * 12=Under Construction
+        * 13=In Service for Test
+        * 14=In Service Commercial
+        * 15=Partial In-Service
+    """
+    iso_df["is_actionable"] = (iso_df["S"].ge(6) & iso_df["S"].lt(11)).fillna(False)
+    iso_df["is_nearly_certain"] = iso_df["S"].ge(11).fillna(False)
+    assert (
+        ~iso_df[["is_actionable", "is_nearly_certain"]].all(axis=1)
+    ).all(), "Some projects are marked marked actionable and nearly certain."
+    return iso_df
+
+
+def _transform_isone(iso_df: pd.DataFrame) -> pd.DataFrame:
+    """Make isone specific transformations."""
+    iso_df = _create_project_status_classification_from_multiple_columns(
+        iso_df,
+        facilities_study_status_col="Facilities Study Status",
+        system_impact_study_col="System Impact Study Status",
+        ia_col="Interconnection Agreement Status",
+        completed_strings=("Document Posted", "Executed"),
+    )
+    return iso_df
+
+
+def transform(raw_dfs: dict[str, pd.DataFrame]) -> dict[str, pd.DataFrame]:
+    """Clean Grid Status Interconnection Queue data.
+
+    Args:
+        raw_dfs: raw dataframes for each ISO.
+
+    Returns:
+        A dictionary of cleaned Grid Status data queus.
+    """
+    # create one dataframe
+    shared_columns_mapping = {
+        "Actual Completion Date": "actual_completion_date",
+        "Capacity (MW)": "capacity_mw",
+        "County": "county",
+        "Generation Type": "resource",
+        "Interconnecting Entity": "interconnecting_entity",
+        "Interconnection Location": "point_of_interconnection",
+        "Project Name": "project_name",
+        "Proposed Completion Date": "proposed_completion_date",
+        "Queue Date": "queue_date",
+        "Queue ID": "queue_id",
+        "State": "state",
+        "Status": "queue_status",
+        "Summer Capacity (MW)": "summer_capacity_mw",
+        "Transmission Owner": "transmission_owner",
+        "Winter Capacity (MW)": "winter_capacity_mw",
+        "Withdrawal Comment": "withdrawal_comment",
+        "Withdrawn Date": "withdrawn_date",
+        "is_actionable": "is_actionable",
+        "is_nearly_certain": "is_nearly_certain",
+        "iso_region": "iso_region",
+    }
+
+    iso_cleaning_functions = {
```

Comment:
This is more of an enhancement than a requirement, but it'd be nice to capture fuel and tech type separately, like EIA does, for the ISOs that report that info separately. That info would improve our CO2 estimates because we wouldn't have to guess if a gas plant is CC or CT. There is a factor of 4 difference in CO2 production!